### DISCUSSION TITLE
fluid particle position export

### DISCUSSION THREAD

**Discussion by BoyuanTang331** [InteractiveComputerGraphics/SPlisHSPlasH#275] [Q&A]
when I use 'partrio exporter' to export the .bgeo file, which kind information of the fluid particle will be export? the position or velocity (relative or absolute) 

![image](https://github.com/InteractiveComputerGraphics/SPlisHSPlasH/assets/117408630/4e3a905c-250a-4fe6-85de-114616388a7d)

If I want to get the position message, what should I do, is there any method to directly read the position of all fluid particles?

I also tried fluid.getPosition(0) in custom_scene.py , but it seems that no result to print
![image](https://github.com/InteractiveComputerGraphics/SPlisHSPlasH/assets/117408630/09af5576-54e7-4b8a-88d1-2530ccf27ccb)

**Comment by BoyuanTang331** [comment]
update：

I use fluid.getPostion() to print the position of each fluid particle, but it seems that it can only print the init position.

How can I get the particle positions of all simulation timesteps?

![image](https://github.com/InteractiveComputerGraphics/SPlisHSPlasH/assets/117408630/0e621886-c75f-4ed9-95c3-17449dbb66f4)


![image](https://github.com/InteractiveComputerGraphics/SPlisHSPlasH/assets/117408630/c99fb86d-dd6c-4acc-b28b-6b60f8834204)

**Comment by janbender** [comment]
You can export any attribute of a particle by defining "particleAttributes" in the scene file or in the scene configuration.
See: https://splishsplash.readthedocs.io/en/latest/file_format.html#export

In a simulation you can print the attributes by selecting particles and press "i" (info). 

I would recommend to export the particle data to VTK files and import them to ParaView. Here you can also get the attribute info for each particle. 

If you want to do it in Python, take a look at the script "callbacks.py" and add your code to "def time_step_callback():". Then in each simulation step the particle positions are printed.

"getPosition" gives you the current position. Maybe your code is only executed before the simulation. 

Hope that helps.
